src/abacus-ts/websocket/markets.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  isWsMarketUpdateResponses,
+  isWsPerpetualMarketResponse,
+} from '@/types/indexer/indexerChecks';
+import { throttle } from 'lodash';
+
+import { timeUnits } from '@/constants/time';
+
+import { type RootStore } from '@/state/_store';
+import { setAllMarketsRaw } from '@/state/raw';
+
+import { createStoreEffect } from '../lib/createStoreEffect';
+import { Loadable, loadableLoaded, loadablePending } from '../lib/loadable';
+import { MarketsData } from '../rawTypes';
+import { selectWebsocketUrl } from '../socketSelectors';
+import { IndexerWebsocket } from './lib/indexerWebsocket';
+import { IndexerWebsocketManager } from './lib/indexerWebsocketManager';
+import { WebsocketDerivedValue } from './lib/websocketDerivedValue';
+
+function marketsWebsocketValue(
+  websocket: IndexerWebsocket,
+  onChange: (val: Loadable<MarketsData>) => void
+) {
+  return new WebsocketDerivedValue<Loadable<MarketsData>>(
+    websocket,
+    {
+      channel: 'v4_markets',
+      id: undefined,
+      handleBaseData: (baseMessage) => {
+        const message = isWsPerpetualMarketResponse(baseMessage);
+        return loadableLoaded(message.markets);
+      },
+      handleUpdates: (baseUpdates, value) => {
+        const updates = isWsMarketUpdateResponses(baseUpdates);
+        let startingValue = value.data;
+        if (startingValue == null) {
+          // eslint-disable-next-line no-console
+          console.log('MarketsTracker found unexpectedly null base data in update');
+          return value;
+        }
+        startingValue = { ...startingValue };
+        updates.forEach((update) => {
+          if (update.oraclePrices != null) {
```

Comment:
there should be one more key that we check. Might need to ask indexer team, but it is emitted when there are changes to a market status
`status`. Not sure what other properties changes are forwarded through this WS channel.

Reply:
checked with backend, they said the status updates are in the trading message updates.